main.go

```diff
@@ -24,6 +25,7 @@ var (
 	conf               *config.Config
 	currentBackoffStep = uint(0)
 	bundleFieldsMap    = warehouse.BundleFields()
+	beginningOfTime    = time.Date(2015, 1, 1, 0, 0, 0, 0, time.UTC)
```

Comment:
Right now this same constant appears in 2 different packages. It might be better to define it as an exported constant in `schema.go`, from where it can be used both in the main package and in the warehouse package. But that difficulty may go away with my proposal that persisting to local disk be modeled as another warehouse. The code for the `LocalWarehouse` would then reside in the warehouse package, where it would have access to the original constant.

Reply:
I thought about modeling local store as another warehouse. My concern is that there are 10 different methods that Warehouse interface defines, most of which will probably do nothing for local store.

Reply:
That's fairly usual in situations like this; could you be more specific about the nature of your concern? Is there something that outweighs the consistency and code reuse we would from keeping the same cognitive model?

Reply:
I may be mistaken, but looking at the list of Warehouse methods, the only method that would clearly make sense for local store is LoadToWarehouse. (I can see how we could maybe do something interesting with LastSyncPoint and DeleteFile, but even that seems to be a bit of a stretch.) If a class would only implement 1 out of 10 interface methods, is it really a good fit for that interface?

Reply:
You are not mistaken, but there is more argument to be had, and I would prefer that we hash it out f2f. I will put a meeting on the calendar for us tomorrow.